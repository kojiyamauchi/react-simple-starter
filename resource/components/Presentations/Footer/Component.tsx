/*
  Footer Component.
*/

import styled from 'styled-components'
import { breakPoint } from '@/styles/StyleMixins'

type Props = {
  className?: string
}

const FooterComponent: React.VFC<Props> = ({ className }): JSX.Element => (
  <footer className={className}>
    <a href="http://kojiyamauchi.com" className="link-letter" target="_blank" rel="noreferrer">
      Â© {new Date().getFullYear()} All The Small Things, All Rights Reserved.
    </a>
  </footer>
)

const StyledComponent = styled(FooterComponent)`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;

  .link-letter {
    font-family: Lobster, cursive;
    font-size: 1.5em;
    text-align: center;
    cursor: pointer;
    ${breakPoint.aboveLargeMobile` font-size: 2.4em `}
  }
`

export const Footer = StyledComponent
