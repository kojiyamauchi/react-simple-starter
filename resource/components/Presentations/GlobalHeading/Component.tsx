/*
  Global Heading Component.
*/
import styled from 'styled-components'
import { BreakPoint } from '@/styles/StyleMixins'

type Props = {
  className?: string
}

const GlobalHeadingComponent: React.VFC<Props> = ({ className }): JSX.Element => <h1 className={`${className} primary-heading`}>React Simple Starter</h1>

const StyledComponent = styled(GlobalHeadingComponent)`
  width: 100%;
  margin: 0 0 40px 0;
  font-size: 3em;
  font-weight: 300;
  text-align: center;
  ${BreakPoint.largeMobile` font-size: 5.6em `}
`

export const GlobalHeading = StyledComponent
