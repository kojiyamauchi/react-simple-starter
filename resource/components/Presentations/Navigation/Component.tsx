/*
  Navigation Component.
*/

import { Link } from 'react-router-dom'
import styled from 'styled-components'

import { breakPoint } from '@/styles/StyleMixins'
import { pathPrefix } from '~/base.config'

type Props = {
  className?: string
}

const link = [
  { path: `${pathPrefix}/`, letter: 'Primary Page', id: 1 },
  { path: `${pathPrefix}/second`, letter: 'Secondary Page', id: 2 },
  { path: `${pathPrefix}/third`, letter: 'Third Page', id: 3 },
  { path: `${pathPrefix}/fourth`, letter: 'Fourth Page', id: 4 }
]

const navigation = (): JSX.Element[] => {
  return link.map(
    (info): JSX.Element => (
      <li key={info.id}>
        <Link to={`${info.path}`} className="link-letter">
          {info.letter}
        </Link>
      </li>
    )
  )
}

const NavigationComponent: React.FC<Props> = ({ className }): JSX.Element => <ul className={className}>{navigation()}</ul>

const StyledComponent = styled(NavigationComponent)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 0 20px;
  margin: 0 auto;
  ${breakPoint.aboveLargeMobile` width: 60%; `}
  ${breakPoint.aboveLargeMobile` padding: 0; `}

  .link-letter {
    font-size: 1.375em;
    font-weight: 300;
    text-align: center;
    ${breakPoint.aboveLargeMobile` font-size: 2.2em `}
  }
`

export const Navigation = StyledComponent
