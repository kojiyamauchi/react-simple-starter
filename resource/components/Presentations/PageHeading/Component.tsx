/*
  PageHeading Component.
*/

import styled from 'styled-components'

import { breakPoint } from '@/styles/StyleMixins'

type Props = {
  className?: string
  letter: string
  translation?: boolean
}

const PageHeadingComponent: React.FC<Props> = ({ className, letter }): JSX.Element => (
  <h2 className={`${className} secondary-heading`}>This is {letter} Page.</h2>
)

const StyledComponent = styled(PageHeadingComponent)`
  font-size: 3em;
  text-align: center;
  ${(props): string | undefined | false => props.translation && `transform: translateY(-100%);`}
  ${breakPoint.aboveLargeMobile` font-size: 4.8em `}
`

export const PageHeading = StyledComponent
