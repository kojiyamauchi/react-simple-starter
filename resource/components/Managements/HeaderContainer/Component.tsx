/*
  Header Container Component.
*/

import styled from 'styled-components'

import { GlobalHeading } from '@/components/Presentations/GlobalHeading'
import { Navigation } from '@/components/Presentations/Navigation'

type Props = {
  className?: string
}

const HeaderContainerComponent: React.FC<Props> = ({ className }): JSX.Element => {
  return (
    <header className={className}>
      <GlobalHeading />
      <Navigation />
    </header>
  )
}

const StyledComponent = styled(HeaderContainerComponent)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
`

export const HeaderContainer = StyledComponent
