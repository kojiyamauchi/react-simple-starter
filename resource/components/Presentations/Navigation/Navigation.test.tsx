/*
  Navigation Component Unit Test.
*/

import { Navigation } from './Component'
import { render /* ,screen */ } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import '@testing-library/jest-dom/extend-expect'
// import userEvent from '@testing-library/user-event'

describe('Navigation Component Unit Test', () => {
  it('Snapshot Testing', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
