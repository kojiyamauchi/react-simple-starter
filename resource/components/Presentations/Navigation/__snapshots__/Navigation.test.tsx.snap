// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component Unit Test Snapshot Testing 1`] = `
<DocumentFragment>
  <ul
    class="sc-bdvvtL kcRiWP"
  >
    <li>
      <a
        class="link-letter"
        href="/react-simple-starter/"
      >
        Primary Page
      </a>
    </li>
    <li>
      <a
        class="link-letter"
        href="/react-simple-starter/second"
      >
        Secondary Page
      </a>
    </li>
    <li>
      <a
        class="link-letter"
        href="/react-simple-starter/third"
      >
        Third Page
      </a>
    </li>
    <li>
      <a
        class="link-letter"
        href="/react-simple-starter/fourth"
      >
        Fourth Page
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
